src/core/serialization.rs

```diff
@@ -103,6 +103,14 @@ impl ArgminCheckpoint {
         self.name.clone()
     }
 
+    /// Get filename for checkpoint
+    #[inline]
+    fn filename(&self) -> String {
+        let mut filename = self.name();
+        filename.push_str(".arg");
```

Comment:
Maybe it would be even better to store `filename` in the `ArgminCheckpoint` struct and have it set to `name` + `".arg"` when when `set_name` is called (or to `default.arg` when `default()` is called). This method would then just return `self.filename`. That way one would also profit from your performance improvements when using the checkpointing mechanism.

Reply:
I was considering this, but didn't, in case a consumer called `set_name()` in the meantime. On reflection, I could simply update the filename inside of `set_name()` to mitigate that. Will update 👍🏼